### DISCUSSION TITLE
mixed effect in dist_permanova

### DISCUSSION THREAD

**Discussion by jorondo1** [david-barnett/microViz#108] [Q&A]
Using the dist_calc() psExtra output, is it possible to pass a restricted permitation design to adonis2 via dist_permanova? such as suggested [here](https://github.com/vegandevs/vegan/issues/429#issuecomment-1260724063)

in adonis2 it would look like this:
```
h1 <- with(df, how(nperm= 9999, blocks = partID)) 
adonis2(dissMx ~ explVar, data = df, permutations = h1, method = "aitchison")
``` 

how would I pass the permutation design to dist_permanova?

**Comment by david-barnett** [comment]
Hi Jonathan

Hmm, interesting question, I have not worked yet with restricted permutation designs, and dist_permanova was not created with this in mind.

I would recommend extracting the distance matrix and sample data from the psExtra object and calling adonis2 directly.
You could use dist_get and samdat_tbl for this, after dist_calc. Beware you will have to handle missing data yourself, excluding any samples with missings in the explanatory variables. You could more easily do this to the phyloseq before running dist_calc, rather than after.

Curious to hear about your study design, why do you need the restricted permutations? I might add support for common situations to dist_permanova in the future 

cheers,
David

**Comment by jorondo1** [comment]
Hi again, thanks for the reply and sorry for posting these as issues!

We have a design where participants provided saliva samples at different points in time. My understanding (from [this issue](https://github.com/vegandevs/vegan/issues/429)) is that restricting permutations to between same-individual samples is equivalent to accounting for participant random effect.

I'll try what you are suggesting, thanks a lot!
